src/MergeContext.js

```diff
@@ -954,12 +968,47 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // recalculates this._stagingFailed value for PRs without staging commit
+    async _recalculateStagingFailed() {
+        assert(!this._stagedSha());
+
```

Comment:
Perhaps we should confirm the intent to recalculate by erasing the old value?

```suggestion

    this._stagingFailed = null;

```

Should not we also init with `null` in the class constructor because, at that time, we do not know the correct value?